modules/tls.go

```diff
@@ -64,5 +59,17 @@ func (s *TLSScanner) InitPerSender(senderID int) error {
 }
 
 func (s *TLSScanner) Scan(t zgrab2.ScanTarget) (zgrab2.ScanStatus, interface{}, error) {
-	return zgrab2.SCAN_SUCCESS, s, nil
+	tcpConn, err := t.Open(&s.config.BaseFlags)
+	if err != nil {
+		return zgrab2.TryGetScanStatus(err), &zgrab2.TLSLog{}, err
```

Comment:
IIRC setting result == nil is the only way to unambiguously communicate that the protocol was not detected (non-success status isn’t enough, since you could get an error collecting optional data after detecting).

But, that means you cannot return e.g. partial TLS handshakes (or even complete handshakes if there is a subsequent failure) — so we may want to revisit that.

Having said that, several other modules do the same thing, so this can probably be addressed with everything else.